pandas/io/excel/_base.py

```diff
@@ -489,11 +493,30 @@ def parse(
         thousands=None,
         comment=None,
         skipfooter=0,
-        convert_float=True,
+        convert_float=None,
         mangle_dupe_cols=True,
         **kwds,
     ):
 
+        if convert_float is None:
```

Comment:
this is slightly magical. can we instead have a function that shows the deprecation and just call it where appropriate so can set the stacklevel w/o this?

Reply:
I don't see how this would get around needing logic to set the stacklevel. Maybe you're asking to move the logic to the new function?

Reply:
@jreback @rhshadrach I've consolidated both stack inspections from _base.py into a single function. It seems like an improvement to me, but let me know if you see other adjustments I should make.

Reply:
After rereading @jreback's comment, it now makes sense to me. The request is to move the deprecation itself to a function (not the determination of stacklevel), and call this further up in the call stack where we don't need to have any logic to determine the stacklevel. If this is doable, I agree it's a better approach.

Reply:
I think this would work: move the `warnings.warn` to a function, call in both read_excel and ExcelFile.parse when appropriate, but in read_excel after emitting the warning change the value of convert_float. Does that make sense?

Reply:
you might be able to use this: `from pandas.util._exceptions import find_stack_level`

Reply:
Oh nice. I looked at `pandas.util._exceptions.find_stack_level ` when I made my last edit, but at the time it seemed specific to "astype" et al. so I tried making something similar. The edits from #41560 two days ago look great. I'll give it a try right away. (I couldn't see a clean way without stack inspection to raise a single warning from both read_excel and ExcelFile.parse. The latter isn't really advertised in the documentation, but it should still raise the correct warnings.)